components/ModHealthcareProfessionalSearchbar.vue

```diff
@@ -9,34 +9,138 @@
+const moderationScreenStore = useModerationScreenStore()
+const facilitiesStore = useFacilitiesStore()
 
 const requestedHealthcareProfessionalId: Ref<string> = ref('')
 // `healthcareProfessionalById` will be used in the UI component in an upcoming PR
-const healthcareProfessionalById: Ref<HealthcareProfessional[] | []> = ref([])
+const healthcareProfessionalByIdOrName: Ref<HealthcareProfessional[] | []> = ref([])
+
+const addHealthcareProfessional = (id: string) => {
+    if (moderationScreenStore.activeScreen === ModerationScreen.EditFacility) {
+        //Check if the healthcareProfessional has already been added
+        const foundProfessionalAlreadyPartOfFacility
+        = facilitiesStore.facilitySectionFields.healthProfessionalsRelations
+            .find(relation => relation.otherEntityId === id)
+            || facilitiesStore.selectedFacilityData?.healthcareProfessionalIds.includes(id)
 
-async function searchHealthcareProfessionalById() {
+        // Return out of the function if already added
+        if (foundProfessionalAlreadyPartOfFacility) return
+
+        // Add a relationship to the facility with selected healthcare professional
+        facilitiesStore.facilitySectionFields.healthProfessionalsRelations.push({
+            action: RelationshipAction.Create,
+            otherEntityId: id
+        })
+
+        // Find the healthcare professional by ID
+        const foundProfessional = healthcareProfessionalsStore.healthcareProfessionalsData
+            .find(healthcareProfessional => healthcareProfessional.id === id)
+
+        // Only push to healthProfessionalsRelationsForDisplay if the professional exists
+        if (foundProfessional) {
+            facilitiesStore.healthProfessionalsRelationsForDisplay.push(foundProfessional)
+            requestedHealthcareProfessionalId.value = ''
+            healthcareProfessionalByIdOrName.value = []
+        }
+    }
+}
+
+function searchHealthcareProfessionalByIdOrName() {
     if (!requestedHealthcareProfessionalId.value || requestedHealthcareProfessionalId.value.trim().length < 1) {
-        return
+        return healthcareProfessionalByIdOrName.value = []
     }
-    healthcareProfessionalById.value = await getHealthcareProfessionalById(requestedHealthcareProfessionalId.value)
+
+    //Search for Ids or names of existing Healthcare professional
+    const healthcareProfessionalByIdOrNamesFound = healthcareProfessionalsStore.healthcareProfessionalsData
+        .filter(healthcareProfessional => {
+            const caseInsensitiveIdMatch = healthcareProfessional.id
+                .toLowerCase()
+                .startsWith(requestedHealthcareProfessionalId.value.toLowerCase().trim())
+
+            const nameMatches = healthcareProfessional.names.some(name => {
+                const firstNameMatch = name.firstName
+                    .toLowerCase()
+                    .includes(requestedHealthcareProfessionalId.value.toLowerCase())
```

Comment:
it looks like you're comparing the wrong value for the names here. It's checking name matching the id.
.....and as I'm writing this I realized your variable is just id but it's also a name input. Let's change the variable name to be something like `searchValue`